falco/values.yaml

```diff
@@ -1,11 +1,149 @@
 # Default values for Falco.
 
+###############################
+# General deployment settings #
+###############################
+
 image:
-  registry: docker.io
-  repository: falcosecurity/falco
-  tag: 0.32.0
   pullPolicy: IfNotPresent
-  pullSecrets: []
+  registry: docker.io
+  repository: falcosecurity/falco-no-driver
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: ""
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+rbac:
+  # Create and use rbac resources
+  create: true
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+# Add additional pod annotations
+podAnnotations: {}
+
+# Add additional pod labels
+podLabels: {}
+
+# Set pod priorityClassName
+podPriorityClassName:
+
+# Set securityContext for the pods
+podSecurityContext: {}
+
+# Set securityContext for the DaemonSet/Deployment containers
+# This block works in combination with the `leastPrivileged` block above.
+securityContext: {}
+
+scc:
+  # true here enabled creation of Security Context Constraints in Openshift
+  create: true
+
+resources:
+  # Although resources needed are subjective on the actual workload we provide
+  # a sane defaults ones. If you have more questions or concerns, please refer
+  # to #falco slack channel for more info about it
+  requests:
+    cpu: 100m
+    memory: 512Mi
+  limits:
+    cpu: 1000m
+    memory: 1024Mi
+
+nodeSelector: {}
+
+affinity: {}
+
+# Allow Falco to run on Kubernetes 1.6 masters.
+tolerations:
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+
+healthChecks:
+  livenessProbe:
+    initialDelaySeconds: 60
+    timeoutSeconds: 5
+    periodSeconds: 15
+  readinessProbe:
+    initialDelaySeconds: 30
+    timeoutSeconds: 5
+    periodSeconds: 15
+
+# If is behind a proxy you can set the proxy server
+proxy:
+  httpProxy:
+  httpsProxy:
+  noProxy:
+
+# Set timezone
+timezone:
+
+extra:
+  # Extra environment variables that will be pass onto Falco containers
+  env: {}
+  # Extra command-line arguments
+  args: []
+  # Additional initContainers for Falco pods
+  initContainers: []
+  # Extra volumes to Falco DaemonSet/Deployment
+  volumes: []
+  # Extra volumeMounts to Falco containers
+  volumeMounts: []
+
+#########################
+# Scenario requirements #
+#########################
+
+controller:
+  kind: daemonset
+  daemonset:
+    # Perform rolling updates by default in the DaemonSet agent
+    # ref: https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/
+    updateStrategy:
+      # You can also customize maxUnavailable or minReadySeconds if you
+      # need it
+      type: RollingUpdate
+  deployment:
+    replicas: 1
```

Comment:
Not sure if we really need to export the replicas number in the `values.yaml` file. As far as I understand we deploy falco in `deployment` mode when using the plugins. At the time being we can have only one `instance` of falco ingesting events from plugins. I would remove `replicas` field for now and add it in the future if needed. What do you think?

Reply:
I'm not totally sure, but I've immagined plugin scenarios that require multiple pods. However, I understand your point and partially agree since there's currently no compelling use case. Let's evaluate this again once the PR is close to be ready.